typescript/token/test/HypERC20.t.sol

```diff
@@ -0,0 +1,366 @@
+        console.log("Overhead gas usage: %d", gasBefore - gasAfter);
+    }
+}
+
+contract HypERC20Test is HypTokenTest {
+    using TypeCasts for address;
+    HypERC20 internal erc20Token;
+
+    function setUp() public override {
+        super.setUp();
+
+        localToken = new HypERC20(DECIMALS);
+        erc20Token = HypERC20(address(localToken));
+
+        erc20Token.initialize(
+            address(localMailbox),
+            TOTAL_SUPPLY,
+            NAME,
+            SYMBOL
+        );
+
+        erc20Token.enrollRemoteRouter(
+            DESTINATION,
+            address(remoteToken).addressToBytes32()
+        );
+        erc20Token.transfer(ALICE, 1000e18);
+
+        _enrollRemoteTokenRouter();
+    }
+
+    function testInitialize_revert_ifAlreadyInitialized() public {
+        vm.expectRevert("Initializable: contract is already initialized");
+        erc20Token.initialize(ALICE, TOTAL_SUPPLY, NAME, SYMBOL);
+    }
+
+    function testTotalSupply() public {
+        assertEq(erc20Token.totalSupply(), TOTAL_SUPPLY);
+    }
+
+    function testDecimals() public {
+        assertEq(erc20Token.decimals(), DECIMALS);
+    }
+
+    function testLocalTransfers() public {
+        assertEq(erc20Token.balanceOf(ALICE), 1000e18);
+        assertEq(erc20Token.balanceOf(BOB), 0);
+
+        vm.prank(ALICE);
+        erc20Token.transfer(BOB, 100e18);
+        assertEq(erc20Token.balanceOf(ALICE), 900e18);
+        assertEq(erc20Token.balanceOf(BOB), 100e18);
+    }
+
+    function testRemoteTransfer() public {
+        remoteToken.enrollRemoteRouter(
+            ORIGIN,
+            address(localToken).addressToBytes32()
+        );
+        uint256 balanceBefore = erc20Token.balanceOf(ALICE);
+        _expectRemoteTransferWithEmit(REQUIRED_VALUE, TRANSFER_AMT, 0);
+        assertEq(erc20Token.balanceOf(ALICE), balanceBefore - TRANSFER_AMT);
+    }
+
+    function testRemoteTransfer_invalidAmount() public {
+        vm.expectRevert("ERC20: burn amount exceeds balance");
+        _expectRemoteTransfer(REQUIRED_VALUE, TRANSFER_AMT * 11);
+        assertEq(erc20Token.balanceOf(ALICE), 1000e18);
+    }
+
+    function testRemoteTransfer_withCustomGasConfig() public {
+        _setCustomGasConfig();
+
+        uint256 balanceBefore = erc20Token.balanceOf(ALICE);
+        _expectRemoteTransferAndGas(
+            REQUIRED_VALUE,
+            TRANSFER_AMT,
+            GAS_LIMIT * IGP_GAS_PRICE
+        );
+        assertEq(erc20Token.balanceOf(ALICE), balanceBefore - TRANSFER_AMT);
+    }
+}
+
+contract HypERC20CollateralTest is HypTokenTest {
+    using TypeCasts for address;
+    HypERC20Collateral internal erc20Collateral;
+
+    function setUp() public override {
+        super.setUp();
+
+        localToken = new HypERC20Collateral(address(primaryToken));
+        erc20Collateral = HypERC20Collateral(address(localToken));
+
+        HypERC20Collateral(address(localToken)).initialize(
+            address(localMailbox)
+        );
```

Comment:
this is the same as erc20collateral.intialize?
imo its easier to understand if you just remove the alias

Reply:
not opposed to it